### ISSUE TITLE
--cache-ttl seems doesn't work

### ISSUE THREAD

**Issue by Sherman-Yang** [GoogleContainerTools/kaniko#896]
**Actual behavior**
I set --cache-ttl=8760h, which is about one year. but Kaniko shows cannot find from cache when I build in the second day for the same project no any changes. what's max value for --cache-ttl ?

**Expected behavior**

**To Reproduce**
1. ...
2. ...

**Additional Information**
 - Dockerfile 
   Please provide either the Dockerfile you're trying to build or one that can reproduce this error.
 - Build Context
   Please provide or clearly describe any files needed to build the Dockerfile (ADD/COPY commands)
 - Kaniko Image (fully qualified with digest)
 
 **Triage Notes for the Maintainers**
 

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [ ] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [ ] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

**Comment by cvgw** [comment]
@Sherman-Yang `--cache-ttl` is a `time.Duration` so its resolution is the same as `int64`

**Comment by Sherman-Chata** [comment]
thank you @cvgw. I find it --cache-ttl works. just sometimes, the hash value is different. so Kaniko cannot find the cache. don't know Kaniko calculate the hash value

**Comment by cvgw** [comment]
@Sherman-Chata yes, that's correct. There are a few open bugs around caching at the moment. There are some PRs in review to address them so hopefully will be fixed soon.

Please let me know if there is anything I can do to help

**Comment by cvgw** [comment]
@Sherman-Chata can you give this another try with v0.15.0?

**Comment by ypadlyak** [comment]
@cvgw v1.16.0 seems to be broken, we set -cache-ttl=168h , but it still fails to find cache
